cmd/osbuild-worker/main.go

```diff
@@ -55,32 +56,33 @@ func createTLSConfig(config *connectionConfig) (*tls.Config, error) {
 	}, nil
 }
 
-func newConnection(remoteAddress string, conf *tls.Config) (net.Conn, error) {
-	if remoteAddress != "" {
-		return tls.Dial("tcp", remoteAddress, conf)
+func NewClient(remoteAddress string, conf *tls.Config) *ComposerClient {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: conf,
+		},
 	}
 
-	// plain non-encrypted connection
-	return net.Dial("unix", "/run/osbuild-composer/job.socket")
+	var scheme string
```

Comment:
What's the logic here? When are we meant to use http and when https, and why was it ok to hardcode to http before this patch?

Reply:
We hard-coded the tls config when connecting before. One of the patches changes that to set the config on Transport and let the library to its magic. It only does that magic for `https`, though. I think it's the more correct thing to do regardless.

Would you like to have this explained in a commit message? Sorry I forgot that.

Reply:
Ah, yes, of course. Thanks!